Implement FIFO on Windows using Named Pipes

Robyt3 opened the pull request:
Reimplement the Linux FIFO file server and client controls on Windows by using Named Pipes.

The DDNet server/client acts as a named pipe server and receives messages.
Messages can be posted to the named pipe server by connecting to it as a client.
The named pipe client can for instance be controlled from the command line with PowerShell.

The PowerShell script `scripts/send_named_pipe.ps1` is added for this purpose.
For example the PowerShell command `./send_named_pipe.ps1 "testpipe" "echo a"` sends the command `echo a` to the pipe named `testpipe`.
Multiple commands can be sent at the same time by separating them with semicolons or newlines.

## Checklist

- [X] Tested the change ingame
- [ ] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test (especially base/) or added coverage to integration test
- [ ] Considered possible null pointers and out of bounds array indexing
- [ ] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

def- commented on src/engine/shared/config_variables.h:
```diff
@@ -356,7 +356,7 @@ MACRO_CONFIG_INT(SvVotePause, sv_vote_pause, 1, 0, 1, CFGFLAG_SERVER, "Allow vot
 MACRO_CONFIG_INT(SvTuneReset, sv_tune_reset, 1, 0, 1, CFGFLAG_SERVER, "Whether tuning is reset after each map change or not")
 MACRO_CONFIG_STR(SvResetFile, sv_reset_file, 128, "reset.cfg", CFGFLAG_SERVER, "File to execute on map change or reload to set the default server settings")
-MACRO_CONFIG_STR(SvInputFifo, sv_input_fifo, 128, "", CFGFLAG_SERVER, "Fifo file to use as input for server console")
+MACRO_CONFIG_STR(SvInputFifo, sv_input_fifo, 128, "", CFGFLAG_SERVER, "Fifo file (Linux) or Named Pipe (Windows) to use as input for server console")
```
Fifo also works on any other UNIX-like OS, including BSD, macOS

Robyt3 replied:
I changed it to `non-Windows`.

def- commented on scripts/send_named_pipe.ps1:
```diff
@@ -0,0 +1,31 @@
+$Wrapper.Reader = New-Object System.IO.StreamReader($Wrapper.Pipe)
+$Wrapper.Writer = New-Object System.IO.StreamWriter($Wrapper.Pipe)
+$Wrapper.Writer.AutoFlush = $true
+$Wrapper.Writer.WriteLine($args[1])
```
If you have more than 2 arguments they will be cut off?

Robyt3 replied:
I changed it so the following arguments are also sent to the pipe.

heinrich5991 commented:
Probably best to drop the #ifdefs around `CFifo`. Less code and we don't support non-unix non-Windows operating systems anyway.

def- commented:
Haiku might be the only candidate.

Robyt3 commented:
I removed the `#ifdefs` around `CFifo`. I also added a loop in `CFifo::Update`, so all messages are read from the pipe every time. And I fixed a memory leak when reading the data fails.

Chairn commented:
A big step forward for https://github.com/ddnet/ddnet/issues/5691, thanks Robyt3

def- commented:
bors r+
Thanks!
